### PR TITLE
code Challenge 1, update angular 15, fix unit test, new end point on …

### PR DESCRIPTION
**Pull request by rreyes02**
…backend to get user-albums, new tab on UI to display records.

### CODE REVIEW

File: backend/UnicornRewards.API/Controllers/UserAlbumController.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                             Id = u.Id,
+                             Name = u.Name,
+                             Email= u.Email,
+                             AlbumName = ua.Where(x => x.UserId == u.Id).Select(y => y.Title).ToList(),
+                         };
+            
+            return Ok(result);
+        }
+
+        private async Task<string> GetAllUsers()
```
Comment (jrodrigoav) [inline backend/UnicornRewards.API/Controllers/UserAlbumController.cs:39]: This should be an HttpClientService

File: backend/UnicornRewards.API/Controllers/UserAlbumController.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                }
+                return result;
+            }
+            catch (Exception)
+            {
+                return string.Empty;
+            }
+        }
+
+        private async Task<string> GetAllAlbums()
```
Comment (jrodrigoav) [inline backend/UnicornRewards.API/Controllers/UserAlbumController.cs:58]: This one goes in to that same service mentioned above

File: backend/UnicornRewards.API/UnicornRewards.API.csproj [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Identity.Web" Version="1.25.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (jrodrigoav) [inline backend/UnicornRewards.API/UnicornRewards.API.csproj:11]: Use the .NET Json serializers instead

File: frontend/src/app/pages/users-albums/users-albums.component.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<div class="container">
+    <div class="row">
+        <div class="col">
+            <ng-container *ngIf="userAlbumsArray.length == 1; else data">
+                <div>No Hay registros para mostrar</div>
```
Comment (jrodrigoav) [inline frontend/src/app/pages/users-albums/users-albums.component.html:5]: This is very nice UX

**Review by jrodrigoav — CHANGES_REQUESTED** [review]
Minor changes, looks good especially the front

File: backend/UnicornRewards.API/Controllers/UserAlbumController.cs (hunk 1 above)
Comment (rreyes02) [reply backend/UnicornRewards.API/Controllers/UserAlbumController.cs]: Solved on the last push

File: backend/UnicornRewards.API/Controllers/UserAlbumController.cs (hunk 2 above)
Comment (rreyes02) [reply backend/UnicornRewards.API/Controllers/UserAlbumController.cs]: Solved on last push

File: backend/UnicornRewards.API/UnicornRewards.API.csproj (hunk 3 above)
Comment (rreyes02) [reply backend/UnicornRewards.API/UnicornRewards.API.csproj:11]: Solved on last push

File: backend/UnicornRewards.API/Services/UserAlbumService.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
+﻿using System.Text.Json;
+using UnicornRewards.API.Models;
+
+namespace UnicornRewards.API.Services
+{
+    public class UserAlbumService
```
Comment (jrodrigoav) [inline backend/UnicornRewards.API/Services/UserAlbumService.cs:6]: Getting there you need to use HttpClientFactory

**Review by jrodrigoav — CHANGES_REQUESTED** [review]
Reach out if you have any questions

File: backend/UnicornRewards.API/Services/UserAlbumService.cs (hunk 5 above)
Comment (jrodrigoav) [reply backend/UnicornRewards.API/Services/UserAlbumService.cs]: Still have the service added twice to the ServiceCollection

**Comment by jrodrigoav** [comment]
Can remove the pdf code now

**Comment by jrodrigoav** [comment]
stale PR, closing so a new one can be opened

**Pull request closed without merging**